src/content-helpers.ts

```diff
@@ -107,35 +111,76 @@ export function makeEmoteMessage(body: string) {
     };
 }
 
+/** Location content helpers */
+
+export const getTextForLocationEvent = (
+    uri: string,
+    assetType: LocationAssetType,
+    timestamp: number,
+    description?: string,
+): string => {
+    const date = `at ${new Date(timestamp).toISOString()}`;
+    const assetName = assetType === LocationAssetType.Self ? 'User' : undefined;
+    const quotedDescription = description ? `"${description}"` : undefined;
+
+    return [
+        assetName,
+        'Location',
+        quotedDescription,
+        uri,
+        date,
+    ].filter(Boolean).join(' ');
+};
+
 /**
  * Generates the content for a Location event
- * @param text a text for of our location
  * @param uri a geo:// uri for the location
  * @param ts the timestamp when the location was correct (milliseconds since
  *           the UNIX epoch)
  * @param description the (optional) label for this location on the map
  * @param asset_type the (optional) asset type of this location e.g. "m.self"
+ * @param text optional. A text for the location
  */
-export function makeLocationContent(
-    text: string,
+export const makeLocationContent = (
     uri: string,
-    ts: number,
+    timestamp?: number,
     description?: string,
     assetType?: LocationAssetType,
-): ILocationContent {
+    text?: string,
+): LegacyLocationEventContent & MLocationEventContent => {
```

Comment:
changing this signature in a non-backwards-compatible way would be a breaking change